### ISSUE TITLE
influxdb_logs sink does not send an HTTP body in 0.11.1

### ISSUE THREAD

**Issue by zackangelo** [timberio/vector#5632]
### Vector Version
```
0.11.1
```

### Vector Configuration File
```toml
data_dir = "/var/run/vector"

[sources.in]
  type = "file" # required
  include = ["/var/run/thermopolis/*.log"] #must match data_dir in thermopolis

#[sources.vector-log]
#  type = "file" # required
#  include = ["/var/log/vector.log"]

#[sources.thermopolis-log]
#  type = "file" # required
#  include = ["/Users/eight/thermopolis/logs/thermopolis.log"]

# lifts and parses the JSON payload out of the message field
# and makes them top level fields
[transforms.json_transform]
  type = "json_parser"
  inputs = ["in"]
  drop_field = true
  drop_invalid = true
  field = "message"

# InfluxDB
[sinks.influxdb]
 inputs = ["json_transform"]
 type = "influxdb_logs"
 endpoint = "https://us-central1-1.gcp.cloud2.influxdata.com"
 namespace = "plc"
 bucket = "plc_mask_counts"
 org = "<REDACTED>"
 token = "<REDACTED>"
 encoding.only_fields = ["timestamp", "machine", "mask_count_track_1", "mask_count_track_2", "employee_badge_id"]
 tags = ["machine", "employee_badge_id"]
```

### Expected Behavior

Vector successfully sends data to InfluxDB. 

### Actual Behavior

Works fine in 0.10.0. After upgrading to 0.11.1, I started receiving the error "points required" from the InfluxDB server in the logs. After turning on debug logging, it appears that Vector was not writing an HTTP body in its request. Reverting back to 0.10.0 fixed the problem.  

### Example Data

```
{"employee_badge_id":null,"file":"/var/run/thermopolis/juliette.log","host":"thermopolis","machine":"juliette","source_type":"file","timestamp":"2020-12-19T22:30:20.622743278Z","track_1_pusher":472550,"track_2_pusher":628112}
```

**Comment by juchiast** [comment]
Hi @zackangelo, how frequent was the error?

**Comment by juchiast** [comment]
I can reproduce the bug
```
Dec 21 14:21:40.170 ERROR sink{component_kind="sink" component_name=influxdb component_type=influxdb_logs}:request{request_id=0}: vector::sinks::util::sink: Response wasn't successful. response=Response { status: 400, version: HTTP/1.1, headers: {"date": "Mon, 21 Dec 2020 07:21:40 GMT", "content-type": "application/json; charset=utf-8", "content-length": "54", "connection": "keep-alive", "x-platform-error-code": "invalid", "strict-transport-security": "max-age=15724800; includeSubDomains"}, body: b"{\"code\":\"invalid\",\"message\":\"writing requires points\"}" }
```

**Comment by juchiast** [comment]
Hi @zackangelo, the data does not have `mask_count_track_1` and `mask_count_track_2`, but rather `track_1_pusher`, `track_2_pusher`. It should work if you change `only_fields` to use `track_1_pusher` and `track_2_pusher` instead.

Vector 0.10 did not apply encoding rules, this was a bug and was fixed, so in vector 0.11 those fields were removed and the event does not have any field left to write to Influxdb.

@binarylogic I'm surprised that vector tried to send an empty buffer, I'll look in to this. I'll also add a warning that an event was dropped.

**Comment by juchiast** [comment]
I will close this, please reopen if you still have the problem.
